pandas/core/groupby/ops.py

```diff
@@ -1085,6 +1085,9 @@ def groups(self):
         }
         return result
 
+    def __iter__(self) -> Iterator[Hashable]:
+        return iter(self.groupings[0].grouping_vector)
```

Comment:
why is the parent class method wrong?

Reply:
Sorry, I forgot to add description. Please, see the description of the PR.

Reply:
I wonder if it would be better to refactor the indices here to align better with the base class rather than override iter. But I'm not certain if it's a good idea and I'm okay with this fix. cc @jbrockmendel

Reply:
@jbrockmendel - friendly ping.

Reply:
Sorry for the delay.  i had in mind something more like @rhshadrach's suggestion, that needing this here might indicate that the self.indices (which the parent class method iterates over) might be wrong